simulatingrisk/hawkdovemulti/model.py

```diff
@@ -194,6 +212,52 @@ def __init__(
         # store whether to compare cumulative payoff or since last adjustment round
         self.adjust_payoff = adjust_payoff
 
+    def _risk_level_in_bounds(self, value):
+        # check if a generated risk level is within bounds
+        return self.min_risk_level <= value <= self.max_risk_level
+
+    def get_risk_attitude_generator(self):
+        """return a generator that will return risk attitudes for individual
+        agents based on the configured distribution."""
+        if self.risk_distribution == "uniform":
+            # uniform/random: generate random integer within risk level range
+            while True:
+                yield self.random.randint(self.min_risk_level, self.max_risk_level)
+        if self.risk_distribution == "normal":
+            # return values from a normal distribution centered around 4
+            while True:
+                yield int(self.random.gauss(4, 1.5))
+        elif self.risk_distribution == "skewed left":
+            # return values from a triangler distribution centered around 0
+            while True:
+                yield round(
+                    self.random.triangular(self.min_risk_level, self.max_risk_level, 0)
+                )
+        elif self.risk_distribution == "skewed right":
+            # return values from a triangler distribution centered around 8
+            while True:
+                yield round(
+                    self.random.triangular(self.min_risk_level, self.max_risk_level, 8)
+                )
```

Comment:
Very nice. I did not know about `random.triangular`. Looks like just what you need here, no more nor less.

Reply:
Again, I think these should be centered around 0 and 9, or around 1 and 8.